testing/unittests/jlibtests.cpp

```diff
@@ -3731,6 +3732,58 @@ class PTreeSerializationDeserializationTest : public CppUnit::TestFixture
     {
         performRoundTripTest(__func__, createBinaryDataCompressionTestPTree(testXml));
     }
+
+    void testCreatePTreeWithNodeCreator()
+    {
+        // Test the createPTree(IBufferedSerialInputStream &src, const char *tag, IPTreeNodeCreator *nodeCreator) function (line 4531)
+
+        // Test data: Create a simple tree to serialize (with case insensitive flag)
+        Owned<IPropertyTree> originalTree = createPTree("TestRoot");
+        originalTree->setProp("@version", "1.0");
+        originalTree->setProp("value", "test content");
+        IPropertyTree *child = originalTree->addPropTree("child");
+        child->setProp("@name", "child1");
+        child->setProp("data", "child content");
+
+        // Serialize the tree to a buffer
+        MemoryBuffer serializedData;
+        originalTree->serialize(serializedData);
+
+        // Create a buffered input stream from the serialized data
+        Owned<IBufferedSerialInputStream> inputStream = createBufferedSerialInputStream(serializedData);
+
+        // Test 1: Call with null nodeCreator (should fall back to createPTree(src, ipt_none))
+        Owned<IPropertyTree> deserializedWithNull = createPTree(*inputStream, "NewTag", nullptr);
+        CPPUNIT_ASSERT(areMatchingPTrees(originalTree, deserializedWithNull));
```

Comment:
this should fail, not sure why it doesn't. Original tree's root is "TestRoot", where as (because of the incorrect passing of a tag to createPTree deserializor), this one is "NewTag". Shouldn't match... not sure how passes.